sdk/identity/Azure.Identity/samples/ClientCertificateCredentialSamples.md

```diff
@@ -0,0 +1,133 @@
+
+
+## Loading certificates from disk
+
+Applications commonly need to load a client certificate from disk. In this case the application can either construct the `ClientCertificateCredential` by specifying the applications tenant id, client id, and the path to the certificate.
+
+```C# Snippet:Identity_CertificateCredenetial_CreateWithPath
+var credential = new ClientCertificateCredential(tenantId, clientId, "./certs/cert.pfx");
+```
+Alternatively the application can construct the `X509Certificate2` themselves such as in this example where the certificate key is password protected.
+
+```C# Snippet:Identity_CertificateCredenetial_CreateWithX509Cert
+var certificate = new X509Certificate2("./certs/cert-password-protected.pfx", "password");
+
+var credential = new ClientCertificateCredential(tenantId, clientId, certificate);
+```
+
+## Loading certificates from an X509Store
+
+Applications running on platforms which provide a secure certificate store might prefer to store and retrieve certificates from there. While the `ClientCertificateCredential` doesn't directly provide a mechanism for this, the application can retrieve the appropriate certificate from the store and use it to construct the `ClientCertificateCredential`. 
+
+Consider the scenario where a pinned certificate used for development authentication is stored in the Personal certificate store. Since the certificate is pinned it can be identified by its thumbprint, which the application might read from configuration or the environment.
+
+```C# Snippet:Identity_CertificateCredenetial_CreateFromStore
+using var store = new X509Store(StoreName.My, StoreLocation.LocalMachine);
+
+store.Open(OpenFlags.ReadOnly);
+
+var certificate = store.Certificates.Cast<X509Certificate2>().FirstOrDefault(cert => cert.Thumbprint == thumbprint);
+
+var credential = new ClientCertificateCredential(tenantId, clientId, certificate);
+```
+
+## Rolling Certificates
+
+Long running applications may have the need to roll certificates during process execution. Certificate rotation is not currently supported by the `ClientCertficateCredential` which treats the certificate used to construct the credential as immutable. This means that any clients constructed with an `ClientCertificateCredential` using a particular cert would fail to authenticate requests after that cert has been rolled and the original is no longer valid. 
+
+However, if an application wants to roll this certificate without creating new service clients, it can accomplish this by creating it's own `TokenCredential` implementation which wraps the `ClientCertificateCredential`. The implementation of this custom credential `TokenCredential` would somewhat depend on how the application handles certificate rotation.
```

Comment:
```suggestion
However, if an application wants to roll this certificate without creating new service clients, it can accomplish this by creating its own `TokenCredential` implementation which wraps the `ClientCertificateCredential`. The implementation of this custom credential `TokenCredential` would somewhat depend on how the application handles certificate rotation.
```